twisted/internet/test/test_socket.py

```diff
@@ -91,6 +96,7 @@ def test_invalidAddressFamily(self):
         reactor = self.buildReactor()
 
         port = socket.socket()
+        port.bind(("127.0.0.1", 0))
```

Comment:
Yes it is necessary to bind() before calling listen() on Windows Python 3.  Windows Python 3 throws an OSError exception with WSA error 10022 (WSAEINVAL) if you don't bind().